scipy/spatial/_spherical_voronoi.py

```diff
@@ -311,9 +311,8 @@ def _calculate_areas_2d(self):
         areas = self.radius * theta
 
         # Correct arcs which go the wrong way (single-hemisphere inputs)
-        signs = np.sign(np.einsum('ij,ij->i', arcs[:, 0],
-                                              self.vertices - self.center))
-        indices = np.where(signs < 0)
+        indices = np.einsum('ij,ij->i', arcs[:, 0],
+                            self.vertices - self.center) < 0
```

Comment:
Just to double-check: this refactoring relies on the fact that `indices` are only used just below in `area[indices]`, and that indexing with a full-size mask (in the new code) is exactly equivalent to indexing with the output of a `nonzero()` (which is what the single-argument where resolves to) 

```
In [6]: x = np.asarray([-1, 0, 1, 2])

In [7]: np.where(np.sign(x) < 0)
Out[7]: (array([0]),)

In [8]: x < 0
Out[8]: array([ True, False, False, False])
```